frontend/src/workspace/funbar/FunBar.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useContext, useEffect } from "react";
+import { Rect, Group, Text } from "react-konva";
+import "../../design/styleSheets/FunBar.css";
+import { Spring, animated } from "react-spring/renderprops-konva";
+import PopupCanvas from "./PopupCanvas";
+import { funBarContext } from "../globals/globals-funbar-dimensions";
+import { globalContext } from "../globals/global-context";
+import { fontContext } from "../globals/globals-fonts";
+
+function FunBar(props) {
+  const width = useContext(globalContext).width;
+  const height = useContext(globalContext).height;
+  const funBarHeight = useContext(globalContext).funBarHeight;
+  const funbarDimensions = useContext(funBarContext);
+  const funBarFontSize = useContext(fontContext).funBarFontSize;
+  const funBarRFFontSize = useContext(fontContext).funBarRFFontSize;
+  // only here because props isn't recognized from Spring
+  const renderFunction = props.renderFunction;
+
+  const [functionButtonHovered, setFunctionButtonHovered] = useState(false);
+  const [imageButtonClicked, setImageButtonClicked] = useState(false);
+  const [imageButtonHovered, setImageButtonHovered] = useState(false);
+
+  return (
+    <Group y={funbarDimensions.funbarY}>
+      {/* <Rect width={50} height={50} fill={'pink'}/>
+      <Text y={100} fill={'blue'} text={funbarDimensions.funbarY}/> */}
+      <BarBase {...props} />
+      <FunctionButton {...props} />
+      <ImageButton {...props} />
+    </Group>
+  );
+
+  function BarBase(props) {
+    return (
+      <Group>
+        {/* <Rect // Blue bar at bottom of workspace
+          x={0}
+          y={0}
+          width={width}
+          height={funBarHeight}
+          fill={props.bg}
+        /> */}
```

Comment:
If this commented block is not needed, you should delete it or leave a reason as to why it is being kept.